### ISSUE TITLE
Improve tool `seq-data-to-lane-bam` to output ubam with required metadata in header

### ISSUE THREAD

**Issue by lindaxiang** [icgc-argo-workflows/dna-seq-processing-tools#121]
Currently tool `seq-data-to-lane-bam` accept both FASTQ and BAM as inputs.
- For FASTQ input,  the metadata are retrieved from the SONG analysis to form the ubam header for each RG
- For BAM input, the output ubam header could carry different information with what are provided in SONG analysis. 

The output ubam are expected to carry all the same information provided in the SONG analysis. So that the downstream tools (bwa-mem, star, hisat2) do not require to have SONG analysis as one of the inputs.

**Comment by lindaxiang** [comment]
A bit of context to add:
For RNA-Seq aligner STAR/HiSAT2, currently the read group information is either derived from the fastq file name or the ubam header (if given as ubam). 

@junjun-zhang what's your thoughts on this thread?

**Comment by junjun-zhang** [comment]
It's a bit complicated here (maybe unnecessarily), but in early development we switched through different ways to pass metdata around in between different steps. Although we have a lot RG fields in the header of **uBAM** (regardless it's generated from original BAM or FASTQ files), the **only** thing matters is RG_ID, which is used at the BWA-MEM alignment step to figure out the proper read_group from the metadata (which contains multiple read groups typically).

For the DNA-Seq alignment workflow, it is at the BWA-MEM alignment step all of the correct metadata information is added to the aligned lane level BAMs. [Here](https://github.com/icgc-argo/dna-seq-processing-tools/blob/bwa-mem-aligner.0.1.12.0/tools/bwa-mem-aligner/bwa-mem-aligner.py#L125-L127) is where `rg_array` is created using the provided metadata input, then later used in `bwa mem` command.

Regarding the RNA-Seq alignment step, it could follow similar idea as the DNA-Seq one, but it will require some code to deal with input metadata parsing. If it's more desireable to avoid the RNA-Seq alignment step to handle metadata, indeed, we could use uBAM header to pass all needed metadata. Looking at the python code for BWA-MEM, we actually made the `metadata` argument optional, when it's omitted, metadat from uBAM header will be used.

@lindaxiang, to answer your quesiton directly, I think what we need to do is to add a reheader step after the `samtools split` step to add all needed RG fields in the uBAM.
